### PR TITLE
Task types

### PR DESCRIPTION
**Pull request by Arczi4**

### CODE REVIEW

File: django_react_proj/users-fe/src/components/TaskLegend/TypesLegend.module.css [hunk 1]
```diff
@@ -0,0 +1,37 @@
+.TypesLegend{
+    box-shadow: rgba(50, 50, 105, 0.15) 0px 2px 5px 0px, rgba(0, 0, 0, 0.05) 0px 1px 1px 0px;
+    width: 20%;
+}
+
+.Meeting {
+    width:10px;
```
Comment (bakowroc) [inline django_react_proj/users-fe/src/components/TaskLegend/TypesLegend.module.css:7]: Those classes can be unified into single class, e.g. `taskTypeCircle` which provides common css attributes and then `Email` `Housework` `Meeting` would provide only color. You can use as many classes as you want

`class="class1 class2"` which for your case would be something like 
```
className={`${styles.taskTypeCircle} ${styles.Email}`}
```
 which is not that good loking, but you can provide utility function or just use already written one which is: https://www.npmjs.com/package/classnames

File: django_react_proj/users-fe/src/components/UsersListRow/UsersListRow.js [hunk 2]
```diff
@@ -23,12 +23,26 @@ const UsersListRow = ({ user, resetState }) => {
     }
   };
 
+  let rowColor = '';
+  if (taskFlag === true) {
```
Comment (bakowroc) [inline django_react_proj/users-fe/src/components/UsersListRow/UsersListRow.js:27]:
```js

if(taskFlag) {
    rowColor = styles.ButtonTrue;
  } else {
}
[...]

```
no need of `else  if (taskFlag === false) `

**Review by bakowroc — CHANGES_REQUESTED** [review]

File: django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js [hunk 3]
```diff
@@ -47,15 +68,33 @@ const NewUserForm = ({ user, resetState, toggle }) => {
         <Input type="text" name="name" onChange={onNameChange} value={defaultIfEmpty(name)} />
       </FormGroup>
       <FormGroup>
-        <Label for="deadline">Data wykonania:</Label>
+        <Row form>
+          <Col>
+            <Label for="deadline">Data wykonania:</Label>
```
Comment (karolszymonczyk) [inline django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js:73]: keep it in one language

File: django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js [hunk 4]
```diff
@@ -10,13 +10,32 @@ const NewUserForm = ({ user, resetState, toggle }) => {
...
   const onNameChange = (event) => {
     setName(event.target.value);
   };
   const onPasswordChange = (event) => {
     setPassword(event.target.value);
   };
+  const onTaskTypeChange = (event) => {
+    setTaskType(event.target.value);
+  };
   const defaultIfEmpty = (value) => (value === '' ? '' : value);
```
Comment (karolszymonczyk) [inline django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js:39]: I don't get it. It always returns value stored in variable `value`.

File: django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js [hunk 5]
```diff
@@ -47,15 +68,33 @@ const NewUserForm = ({ user, resetState, toggle }) => {
...
+                disableClock="true"
+                name="deadline"
+                onChange={setDeadline}
+                value={defaultIfEmpty(deadline)}
+              />
+            </Col>
+            <Col sm={7}>
+              <Input type="select" onChange={onTaskTypeChange} value={defaultIfEmpty(taskType)}>
+                {types.map((type) => (
+                  <option name="taskType">{type.type}</option>
```
Comment (karolszymonczyk) [inline django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js:93]: components rendered inside `map` should have key attribute which must be unique.

**Review by karolszymonczyk — CHANGES_REQUESTED** [review]
`pycache` should be in gitignore

File: django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js (hunk 4 above)
Comment (Arczi4) [reply django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js:39]: So, it should be just 'const defaultIfEmpty = (value) => value;' ?

File: django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js (hunk 3 above)
Comment (Arczi4) [reply django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js:73]: Just like we talked, Lukasz already changed it.

**Review by bakowroc — APPROVED** [review]

File: django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js (hunk 4 above)
Comment (karolszymonczyk) [reply django_react_proj/users-fe/src/components/NewUserForm/NewUserForm.js]: just `value={deadline}` in `TimePicker` ond `value={raskType}` itd.

**Review by karolszymonczyk — APPROVED** [review]

**Pull request merged**